### ISSUE TITLE
[New Dist Option] Create AppImage for linux

### ISSUE THREAD

**Issue by ParaplegicRacehorse** [feross/webtorrent-desktop#1102]
Not all of us have administrator privileges on the systems we use. [AppImage](https://github.com/probonopd/AppImage) is a way to use software without mucking with system services or requiring administrative rights.

**Comment by probonopd** [comment]
Generated an AppImage based on the deb on the GitHub Releases page:
https://bintray.com/probono/AppImages/WebTorrent/_latestVersion#files

Here is the yml file that generated the AppImage automatically on Travis CI:
https://github.com/probonopd/AppImages/blob/master/recipes/meta/WebTorrent.yml

**Comment by ParaplegicRacehorse** [comment]
Thanks @probonopd.

@feross, what's the chance this can be integrated into your build system?

**Comment by feross** [comment]
How is this different from Snap packages? I'm not familiar with the pros-cons of each. There seems to be quite a bit of overlap in their goals. I'd prefer to pick just one of these "universal" formats to support.

**Comment by probonopd** [comment]
The biggest difference is that AppImages run "out-of-the-box" on most common Linux desktop distributions, without the need for the user to install something beforehand.

Providing an [AppImage](http://appimage.org/) would have, among others, these advantages:
- One app = one file = super simple for users: just download one AppImage file, [make it executable](http://discourse.appimage.org/t/how-to-make-an-appimage-executable/80), and run
- No unpacking or installation necessary
- No root needed
- No system libraries changed
- Just one format for all major distributions
- Works out of the box, no installation of runtimes needed
- Optional(!) desktop integration with `appimaged`
- Binary delta updates, e.g., for continuous builds (only download the binary diff) using AppImageUpdate
- Can GPG2-sign your AppImages (inside the file)

**Comment by ParaplegicRacehorse** [comment]
Right; and snap (and flatpack, for that matter) still requires administrator privileges to install because it installs into the system portion of the filesystem. AppImages can be treated just like MacOS App-packages and thrown any old place the user wants and don't require administrative rights on the machine being installed to.

**Comment by probonopd** [comment]
@ParaplegicRacehorse did you have a chance to try out the example WebTorrent AppImage yet?

**Comment by ParaplegicRacehorse** [comment]
@probonopd, yes. It works as expected, which is to say: it works very well (given my relatively slow internet connection) and I experienced no problems on my UbuntuGNOME 16.04 LTS system.

**Comment by probonopd** [comment]
So, would this project be interested in providing an official one?

**Comment by Serkan-devel** [comment]
Any updates on this?

**Comment by Serkan-devel** [comment]
But what's missing here for upstream release?

**Comment by Yhozen** [comment]
It's a great alternative for people using any distro that is not based on Debian/Ubuntu

**Comment by probonopd** [comment]
Commenting here so that @Stale goes away; no one doing anything is never a good reason for closing valid issues. I'm here, happy to help.

**Comment by Serkan-devel** [comment]
who came to the idea of that bot anyways?

**Comment by Serkan-devel** [comment]
I anyone still here?

**Comment by probonopd** [comment]
Sure! I am here, happy to help, if one of the developers of this projects wants to make an AppImage.

**Comment by Serkan-devel** [comment]
@probonopd has there any .yml-file for packaging been created?

**Comment by probonopd** [comment]
No.

**Comment by Borewit** [comment]
Looks like you guys have solution in mind.
I am not familiar with AppImage format.

The lack of admin rights is not conflicting with the WebTorrent requirements?

Would I be able to test this in om my Fedora Linux system?

Related to #1502 (Linux RPM package)

**Comment by probonopd** [comment]
@Borewit I described the format briefly in https://github.com/webtorrent/webtorrent-desktop/issues/1102#issuecomment-274252209. More details are described at https://docs.appimage.org/introduction/index.html. Kindly let me know if you are interested in making an official AppImage, then I can give you a hand.

**Comment by Borewit** [comment]
Better Linux integration makes sense to me. I am not sure what the best way is.

As a fedora user I expect things to be installable with dnf, only if I have a very specific need I considering to install things outside dnf.

But I know Debian and Ubuntu are also very populair.

I have to admit that I don't have a lot of time. If I see inittiatives taken on this project I try to support it. If I feel confortable reviewing it in a reseanable amount of time I try to do so. There are way more good ideas posted then there are fingers to write them. That is why @feross enabled the stale bot. It doesn't make sense to keep piling up idea's. So if you want to have something to be done, you have to take the initiative yourself. In addition to that, there are good open PR's which did not get the attentation they deserve.

**Comment by probonopd** [comment]
> I have to admit that I don't have a lot of time. 

Sounds familiar.

> there are good open PR's which did not get the attentation they deserve.

Which is exactly why I am asking for the level of interest in an AppImage PR first before I start working on one.

**Comment by strypey** [comment]
I came here hoping there was an AppImage. Anyone who wants to try WebTorrent Desktop on GNU/Linux, but uses a distro that hasn't packaged it in their official repos. would potentially want to use an AppImage. If you do make one, it would be great to make it downloadable. Details on how to do that in this interview with the creator of AppImage: https://itsfoss.com/appimage-interview/

**Comment by probonopd** [comment]
Yes it is still relevant @github-actions.
